llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -868,10 +873,9 @@ void llvm::computeKnownBitsFromContext(const Value *V, KnownBits &Known,
 
     Value *Arg = I->getArgOperand(0);
 
-    if (Arg == V && isValidAssumeForContext(I, Q.CxtI, Q.DT)) {
-      assert(BitWidth == 1 && "assume operand is not i1?");
-      (void)BitWidth;
-      Known.setAllOnes();
+    if (match(Arg, m_TruncOrSelf(m_Specific(V))) &&
+        isValidAssumeForContext(I, Q.CxtI, Q.DT)) {
+      Known.One.setBit(0);
```

Comment:
Should we also handle branch instructions on `(trunc i1)` in the above loop?

Reply:
`(trunc i1)` is not added to DomConditionCache as it is now so maybe leave it for an other PR?

Reply:
I think we already do (in the same place as we add assumes): https://github.com/llvm/llvm-project/blob/main/llvm/lib/Analysis/ValueTracking.cpp#L10160

Reply:
but it is only for assumes that that function is called for value 
https://github.com/llvm/llvm-project/blob/1c7625b350755a1e2873282c125ca91691a1a7d6/llvm/lib/Analysis/ValueTracking.cpp#L10193

Reply:
created https://github.com/llvm/llvm-project/pull/126414 for this.